### PR TITLE
ROC- 5006 Link defendant to claim and perform response journey

### PR DESCRIPTION
**Pull request by dharmendrak**
### JIRA link (if applicable) ###
[ROC-4600](https://tools.hmcts.net/jira/browse/ROC-4600)
[ROC-4601](https://tools.hmcts.net/jira/browse/ROC-4601)
[ROC-5006](https://tools.hmcts.net/jira/browse/ROC-5006)
[ROC-4504](https://tools.hmcts.net/jira/browse/ROC-4504)

### Change description ###
Link defendant
Perform response journey

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ClaimMapper.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
 @Component
 public class ClaimMapper {
```
Comment (karthikkannankainos) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ClaimMapper.java]: There is no test for claimmapper. Needs to be written

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ClaimMapper.java [hunk 2]
```diff
@@ -94,11 +103,13 @@ public void to(Claim claim, CCDCase.CCDCaseBuilder builder) {
 
     public void from(CCDCase ccdCase, Claim.ClaimBuilder claimBuilder) {
         Objects.requireNonNull(ccdCase, "ccdClaim must not be null");
```
Comment (karthikkannankainos) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ClaimMapper.java]: Suggestion... Change the failure message to "ccdCase must be null"

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DefendantRepresentativeMapper.java [hunk 3]
```diff
@@ -26,29 +26,30 @@ public DefendantRepresentativeMapper(
...
-            && isBlank(ccdDefendant.getRepresentativeOrganisationDxAddress())
-            && isBlank(ccdDefendant.getRepresentativeOrganisationEmail())
+        if (isBlank(ccdDefendant.getClaimantProvidedRepresentativeOrganisationName())
+            && ccdDefendant.getClaimantProvidedRepresentativeOrganisationAddress() == null
+            && isBlank(ccdDefendant.getClaimantProvidedRepresentativeOrganisationPhone())
+            && isBlank(ccdDefendant.getClaimantProvidedRepresentativeOrganisationDxAddress())
+            && isBlank(ccdDefendant.getClaimantProvidedRepresentativeOrganisationEmail())
         ) {
             return null;
         }
```
Comment (karthikkannankainos) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DefendantRepresentativeMapper.java]: Representative use the builder pls

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/ClaimantMapperTest.java [hunk 4]
```diff
@@ -54,7 +58,7 @@ public void shouldMapOrganisationToCCD() {
         Party party = SampleParty.builder().organisation();
 
         //when
-        CCDClaimant ccdParty = claimantMapper.to(party);
+        CCDClaimant ccdParty = claimantMapper.to(party, claim);
```
Comment (karthikkannankainos) [inline ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/ClaimantMapperTest.java]: ClaimantMapper has changes 

builder.partyEmail(claim.getSubmitterEmail());

But its not tested. Please add this assertion

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataServiceTest.java [hunk 5]
```diff
@@ -121,7 +123,7 @@ public void before() {
             anyString()
         ))
             .thenReturn(StartEventResponse.builder()
-                .caseDetails(CaseDetails.builder().build())
+                .caseDetails(CaseDetails.builder().data(Maps.newHashMap()).build())
```
Comment (karthikkannankainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataServiceTest.java]: Suggestion : use Collections.<String,Object>emptyMap() instead

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ClaimMapper.java (hunk 1 above)
Comment (dharmendrak) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ClaimMapper.java]: caseMapper test covers tests for this as ccdClaimData is not an object

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ClaimMapper.java (hunk 2 above)
Comment (dharmendrak) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ClaimMapper.java]: done

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DefendantRepresentativeMapper.java (hunk 3 above)
Comment (dharmendrak) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DefendantRepresentativeMapper.java]: done

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/ClaimantMapperTest.java [hunk 6]
```diff
@@ -30,10 +35,11 @@ public void shouldMapIndividualToCCD() {
         Party party = SampleParty.builder().individual();
 
         //when
-        CCDClaimant ccdParty = claimantMapper.to(party);
+        CCDClaimant ccdParty = claimantMapper.to(party, claim);
 
         //then
         assertThat(party).isEqualTo(ccdParty);
+        Assertions.assertThat(ccdParty.getPartyEmail()).isEqualTo(claim.getSubmitterEmail());
```
Comment (sabahirfan) [inline ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/ClaimantMapperTest.java]: Unnecessary use of fully qualified name 'Assertions.assertThat' due to existing static import 'uk.gov.hmcts.cmc.ccd.assertion.Assertions.assertThat'

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataServiceTest.java [hunk 7]
```diff
@@ -207,28 +209,28 @@ public void submitPostPaymentShouldReturnClaim() {
...
-        verify(jsonMapper).convertValue(caseDataCaptor.capture(), eq(CCDCase.class));
+        verify(jsonMapper).fromMap(caseDataCaptor.capture(), eq(CCDCase.class));
         assertEquals(SampleClaim.CLAIM_ID, caseDataCaptor.getValue().get("id"));
     }
 
     @Test
     public void requestMoreTimeForResponseShouldReturnClaim() {
-        Claim providedClaim = SampleClaim.getDefault();
-        Claim expectedClaim = SampleClaim.claim(null, "000MC001");
+        Claim providedClaim = SampleClaim.getWithResponse(null);
```
[the remark this answers is not in the document]
Comment (dharmendrak) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataServiceTest.java]: done

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataServiceTest.java (hunk 5 above)
Comment (dharmendrak) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataServiceTest.java]: it become immutable, and we need adding something

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/ClaimantMapperTest.java (hunk 4 above)
Comment (dharmendrak) [reply ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/ClaimantMapperTest.java]: there is no object call ed ccdClaimData, but a ccdClaim, so assertion is covered in CaseMapperTest

**Pull request merged**